### ISSUE TITLE
ansible-galaxy fails during install

### ISSUE THREAD

**Issue by manuelmoritz1308** [geerlingguy/drupal-vm#482]
Hi,
I upgraded to the latest version of DrupalVM and now I'm not able to run  `ansible-galaxy install`.
I get the following output:

```shell
- downloading role 'php', owned by geerlingguy
https://galaxy.ansible.comhttps://galaxy.ansible.com/api/v1/roles/432/versions/?page=2&page_size=50
Traceback (most recent call last):
  File "/usr/local/Cellar/ansible/1.9.4/libexec/bin/ansible-galaxy", line 959, in <module>
    main()
  File "/usr/local/Cellar/ansible/1.9.4/libexec/bin/ansible-galaxy", line 953, in main
    fn(args, options, parser)
  File "/usr/local/Cellar/ansible/1.9.4/libexec/bin/ansible-galaxy", line 819, in execute_install
    if len(role_versions) > 0:
TypeError: object of type 'NoneType' has no len()
```

According to homebrew the latest ansible version is installed. I'm on Mac OS X 10.11.3.

Thanks for your help.

**Comment by oxyc** [comment]
Seems you are getting the same problem as https://github.com/geerlingguy/drupal-vm/issues/339#issuecomment-190138924. No clue what this is caused by though. Might even be upstream like @luckyape says.

**Comment by geerlingguy** [comment]
Gah, I just ran into this in some testing too. I'm offline most of the morning so can't investigate now, but I'm hoping there's some simple fix, either in the upstream role or in Galaxy...

**Comment by geerlingguy** [comment]
I wonder if this could be an issue with Galaxy and the number of tags there are for the PHP role; can you try once more, as I reimported it on Galaxy. If it's Galaxy, the hack suggested in #339 might be the best short term solution.

**Comment by geerlingguy** [comment]
See also, build failure on Travis: https://travis-ci.org/geerlingguy/ansible-role-php-pecl/builds/112516942

**Comment by geerlingguy** [comment]
Apparently Galaxy 2.0.1 was also released today/last night... I wonder if that has anything to do with this.

**Comment by oxyc** [comment]
I opened an issue upstream https://github.com/ansible/galaxy-issues/issues/134

**Comment by geerlingguy** [comment]
Thanks @oxyc - It looks like this should be fixed tonight. A temporary fix (deleting a few old tags in the repo) didn't seem to work, as they're still in Galaxy.

**Comment by backlineint** [comment]
Not sure if it helps, but the hack in #339 does in fact allow the provisioning to finish.

**Comment by manuelmoritz1308** [comment]
Hi there, thanks for your help so far.
Today I noticed my Homebrew wasn't up to date and so I was working with Ansible 1.9.4. After upgrading to 2.0.1.0 the install finishes without errors. I don't know if this is still an issue for you the, if not, the ticket can be closed.

Thanks for your help.

**Comment by kgrey-ikos** [comment]
I am also having this problem when running `ansible-galaxy install`. My Homebrew says my ansible version of 1.9.2 is the latest version.
I feel this is resulting in an error running `vagrant up` where it fails on `TASK: [geerlingguy.java | Ensure Java is installed.]` error `with_items expects a list or a set`

**Comment by manuelmoritz1308** [comment]
@kgrey-ikos My Homebrew told me the same. You need to `brew update` to get the latest formulas and then `brew outdated` should tell you that ansible is indeed outdated. The run `brew upgrade ansible` and you should have the newest version.

**Comment by oxyc** [comment]
The fix that ansible galaxy implemented only works for Ansible v2 apparently, so for now it is a requirement. I believe the fixes are already merged into Ansible master but it will probably take a while before new releases come out :/

**Comment by brentratliff** [comment]
It failed for me last night but just worked this morning.

**Comment by kgrey-ikos** [comment]
Thanks @manuelmoritz1308 that did the job for me and got `vagrant up` to run without errors. :+1:

**Comment by ccjjmartin** [comment]
@manuelmoritz1308: Thank you so much for your post.  This gave me the answer that was right in front of my face but I couldn't see it.  I needed to upgrade Ansible. Makes sense if they release a new version.  I however didn't use brew to install Ansible, I used pip so my upgrade was:

`sudo pip install ansible --upgrade`
`sudo ansible-galaxy install -r provisioning/requirements.yml --force`
`vagrant up`

Most people probably won't have to do this but I also:
`vagrant destroy` - deleted my vm
`vagrant box update` - updated the base box
`vagrant up`

Downloads of requirements now work but with a fresh box setup (with PHP 7.0 and MariaDB) I received:

TASK [Add repository for PHP 7.0.] *********************************************
fatal: [drupalvm]: FAILED! => {"changed": false, "failed": true, "msg": "failed to fetch PPA information, error was: HTTP Error 404: Not Found"}

Created new issue #491 with a temporary fix of downgrading back to PHP 5.6 which allows the provisioning to finish with MariaDB.

**Comment by geerlingguy** [comment]
I think this is resolved now—the PHP 7.0 issue should be working at this point too.

**Comment by malcomio** [comment]
I also hit the same issue, and after trying to update ansible, I eventually fixed it by editing my local ansible-galaxy file with the changes from https://github.com/ansible/ansible/commit/d2bd6604b0e5611d43e0da991af6b624f038a03d
